src/components/CreateTeam/container.js

```diff
@@ -46,12 +49,6 @@ export default Wrapped =>
         );
     }
 
-    shouldComponentUpdate(nextProps, nextState) {
-      const teamSubmittedhasChanged = nextState.teamSubmitted !== this.state.teamSubmitted;
-      const clientListhasChanged = nextState.clients.length !== this.state.clients.length;
-      return teamSubmittedhasChanged || clientListhasChanged;
-    }
-
     submitRequest = data => {
```

Comment:
Pass the reset form function into here so you can reset the form when the submission was successful.

Reply:
This was so obvious once you said this! removed al state and passing resetForm up to submitRequest.